src/hooks/useGoogleAuth.jsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react';
+const useGoogleAuth = () => {
+  useEffect(() => {
+    window.gapi.load('auth2', () => {
+      window.gapi.auth2.init({
+        clientId:
+          '792255729562-tjcf5k4uv35et7e3v0e1g2pkmqibpsn8.apps.googleusercontent.com',
+      });
+    });
+  }, []);
+
+  const login = () => {
+    return window.gapi.auth2
+      .getAuthInstance()
+      .signIn()
+      .then((user) => {
+        const profile = user.getBasicProfile();
+        const userData = {
+          name: profile.getName(),
+          email: profile.getEmail(),
+          picture: profile.getImageUrl(),
+        };
+        return userData;
```

Comment:
it can be simplified

```suggestion
        return {
          name: profile.getName(),
          email: profile.getEmail(),
          picture: profile.getImageUrl(),
        };
        
```